pkg/reference/ref.go

```diff
@@ -1,11 +1,32 @@
 package reference
 
 import (
-	kapi "k8s.io/api/core/v1"
 	"reflect"
+	"regexp"
 	"strings"
+
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func ObjectKey(obj metav1.Object) client.ObjectKey {
+	return client.ObjectKey{
+		Name:      obj.GetName(),
+		Namespace: obj.GetNamespace(),
+	}
+}
+
+func TruncateName(name string) string {
```

Comment:
I don't understand why this is part of the `ref` package.  It's seems completely unrelated.  Also, it's only used in 1 place.

Reply:
It looks like Danil used it in 3 places, from two different packages. Perhaps that's why he put it under ref. Where should it go?

Reply:
Not sure but putting it here is arbitrary and we've worked hard to keep our packages clean.
Any idea why we need to build _special_ names instead of using labels?

Reply:
Well we don't want a label. We need an object name. Specifically, an input to generateName. For example, here's one use:
```
				GenerateName: migref.TruncateName("stage-"+ref.Name) + "-",
￼```
Creating a stage pod name where ref.Name might be an existing pod name, it might be a DeploymentConfig name, or it might be a PVC name (depending on context).

Reply:
That said, I don't see any reason to use TruncateName where it's used in migplan/registry.go. If we remove that instance, it's only needed in stage.go and it can be an unexported local function.